drivers/volume/portworx/portworx.go

```diff
@@ -356,6 +357,29 @@ func (d *portworx) getPxNode(n node.Node, cManager cluster.Cluster) (api.Node, e
 	return pxnode.(api.Node), nil
 }
 
+func (d *portworx) getPxVersionOnNode(n node.Node, cManager cluster.Cluster) (string, error) {
+
+	pxNode, err := d.getPxNode(n, cManager)
+	if err != nil {
+		return "", err
+	}
+
+	t := func() (interface{}, bool, error) {
```

Comment:
this timeout look doesn't make sense.

In each retry, you are using the same pxNode that you got from a single attempt above. So nothing is going to change between subsequent retries.

Reply:
fixed